### PR TITLE
Access control tests

### PR DESCRIPTION
**Pull request by sebrandon1**
Copied over the following tests from the old suite:
- [It] access-control access-control-pod-role-bindings [access-control-pod-role-bindings]",
- [It] access-control access-control-cluster-role-bindings [access-control-cluster-role-bindings]",
- [It] access-control access-control-pod-automount-service-account-token [access-control-pod-automount-service-account-token]",

### CODE REVIEW

File: cnf-certification-test/accesscontrol/rbac/automount.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+func (at *AutomountToken) AutomountServiceAccountSetOnSA() (ServiceAccountTokenStatus, error) {
+	// Create a default SATS struct.
+	saStatus := ServiceAccountTokenStatus{
+		TokenSet:   false,
+		TokenValue: false,
+	}
+
+	// Collect all of the service accounts that live in the target namespace.
+	saList, saErr := at.ClientHolder.K8sClient.CoreV1().ServiceAccounts(at.podNamespace).List(context.TODO(), v1.ListOptions{})
```
Comment (hamadise) [inline cnf-certification-test/accesscontrol/rbac/automount.go:39]: do you need all of them ? I think you can use Get and provide the one you need. The whole list is not needed

Comment (sebrandon1) [reply cnf-certification-test/accesscontrol/rbac/automount.go:39]: Good point.  I'll check that out.

**Review by hamadise — APPROVED** [review]

File: cnf-certification-test/accesscontrol/suite.go [hunk 2]
```diff
@@ -303,3 +322,132 @@ func TestPodServiceAccount(env *provider.TestEnvironment) {
...
+		rbTester := rbac.NewClusterRoleBindingTester(put.Spec.ServiceAccountName, put.Namespace, clientsholder.NewClientsHolder())
+
+		// Get any clusterrolebindings that do not belong to the pod namespace.
+		clusterRoleBindings, err := rbTester.GetClusterRoleBindings()
+		if err != nil {
+			failedPods = append(failedPods, put.Name)
+		}
+
+		if len(clusterRoleBindings) > 0 {
+			logrus.Warnf("Pod: %s has the following cluster role bindings: %s", put.Name, clusterRoleBindings)
```
Comment (edcdavid) [inline cnf-certification-test/accesscontrol/suite.go:380]: Add failing clusterRoleBindings to the claims file

File: cnf-certification-test/accesscontrol/suite.go [hunk 3]
```diff
@@ -303,3 +322,132 @@ func TestPodServiceAccount(env *provider.TestEnvironment) {
...
+
+		// The pod token is false means the pod is configured properly
+		// The pod is not configured and the service account is configured with false means
+		// the pod will inherit the behavior `false` and the test would pass
+		if !podServiceTokenStatus.TokenValue && !saStatus.TokenValue {
+			continue
+		}
+
+		// the service account is configured with true means all the pods
+		// using this service account are not configured properly, register the error
```
Comment (edcdavid) [inline cnf-certification-test/accesscontrol/suite.go:430]: // the service account is configured with true means not all the pods
		// using this service account are configured properly, register the error
		// message and fail

File: cnf-certification-test/accesscontrol/suite.go [hunk 4]
```diff
@@ -303,3 +322,132 @@ func TestPodServiceAccount(env *provider.TestEnvironment) {
...
+		gomega.Expect(put.Spec.ServiceAccountName).ToNot(gomega.BeEmpty())
+
+		// The token can be specified in the pod directly
+		// or it can be specified in the service account of the pod
+		// if no service account is configured, then the pod will use the configuration
+		// of the default service account in that namespace
+		// the token defined in the pod has takes precedence
+		// the test would pass iif token is explicitly set to false
+		// if the token is set to true in the pod, the test would fail right away
+		podServiceTokenStatus := rbac.DetermineStatus(put.Spec.AutomountServiceAccountToken)
```
Comment (edcdavid) [inline cnf-certification-test/accesscontrol/suite.go:407]: I find the DetermineStatus / Token status object function confusing. Why not just test on (same for sa) :
 - put.Spec.AutomountServiceAccountToken!=nil  -> there is no automountServiceAccountToken param configured
 - put.Spec.AutomountServiceAccountToken == true -> it is configured to true
 - put.Spec.AutomountServiceAccountToken == false -> it is configured to false

File: cnf-certification-test/accesscontrol/rbac/automount.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+)
+
+type AutomountToken struct {
+	serviceAccountName string
+	podNamespace       string
+	ClientHolder       *clientsholder.ClientsHolder
+}
+
+type ServiceAccountTokenStatus struct {
+	TokenSet   bool
```
Comment (edcdavid) [inline cnf-certification-test/accesscontrol/rbac/automount.go:18]: TokenSet   is really isAutomountServiceAccountTokenPresent
TokenValue is really automountServiceAccountTokenValue
I would just remove this object, see below

File: cnf-certification-test/accesscontrol/suite.go (hunk 2 above)
Comment (sebrandon1) [reply cnf-certification-test/accesscontrol/suite.go:381]: Done.

File: cnf-certification-test/accesscontrol/suite.go (hunk 4 above)
Comment (sebrandon1) [reply cnf-certification-test/accesscontrol/suite.go]: Removed DetermineStatus().  Thanks!

File: cnf-certification-test/accesscontrol/rbac/automount.go (hunk 5 above)
Comment (sebrandon1) [reply cnf-certification-test/accesscontrol/rbac/automount.go:34]: Removed DetermineStatus().

File: cnf-certification-test/accesscontrol/suite.go [hunk 6]
```diff
@@ -303,3 +322,134 @@ func TestPodServiceAccount(env *provider.TestEnvironment) {
...
+		saStatus, err := crbTester.AutomountServiceAccountSetOnSA()
+		if err != nil {
+			failedPods = append(failedPods, put.Name)
+			continue
+		}
+
+		// The pod token is false means the pod is configured properly
+		// The pod is not configured and the service account is configured with false means
+		// the pod will inherit the behavior `false` and the test would pass
+		if !*put.Spec.AutomountServiceAccountToken || (saStatus != nil && !*saStatus) {
```
Comment (edcdavid) [inline cnf-certification-test/accesscontrol/suite.go:426]: !*put.Spec.AutomountServiceAccountToken -> 
( put.Spec.AutomountServiceAccountToken != nil && !*put.Spec.AutomountServiceAccountToken )

File: cnf-certification-test/accesscontrol/suite.go [hunk 7]
```diff
@@ -303,3 +322,134 @@ func TestPodServiceAccount(env *provider.TestEnvironment) {
...
+			continue
+		}
+
+		// the service account is configured with true means all the pods
+		// using this service account are not configured properly, register the error
+		// message and fail
+		if saStatus != nil && *saStatus {
```
Comment (edcdavid) [inline cnf-certification-test/accesscontrol/suite.go:433]: suggesting renaming saStatus to saAutomountServiceAccountToken

File: cnf-certification-test/accesscontrol/suite.go (hunk 6 above)
Comment (sebrandon1) [reply cnf-certification-test/accesscontrol/suite.go:426]: Done.

File: cnf-certification-test/accesscontrol/suite.go (hunk 7 above)
Comment (sebrandon1) [reply cnf-certification-test/accesscontrol/suite.go:433]: Done.

File: cnf-certification-test/accesscontrol/suite.go [hunk 8]
```diff
@@ -303,3 +322,134 @@ func TestPodServiceAccount(env *provider.TestEnvironment) {
...
+
+		// Get any rolebindings that do not belong to the pod namespace.
+		roleBindings, err := rbTester.GetRoleBindings()
+		if err != nil {
+			failedPods = append(failedPods, put.Name)
+		}
+
+		if len(roleBindings) > 0 {
+			logrus.Warnf("Pod: %s has the following role bindings: %s", put.Name, roleBindings)
+			tnf.ClaimFilePrintf("Pod: %s has the following cluster role bindings: %s", put.Name, roleBindings)
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply cnf-certification-test/accesscontrol/suite.go]: Done.

**Review by edcdavid — APPROVED** [review]

**Pull request merged**